File: system.pl
#!/bin/perl

use strict;
use warnings;

print "pass-through environment:\n";
system("env > /tmp/perl.thru.env.sh");

print "safe environment: ";
{
  local %ENV = ("PATH" => "/bin");
  system("env > /tmp/perl.safe.env.sh");
}
print "\n";

File: system.py
#!/bin/python2

import os

# os.system always uses the parent's environment
# stop using it.
print("pass-through environment: ");
os.system("env > /tmp/python.thru.env.sh")

print "safe environment: ";
from subprocess import call
call("env > /tmp/python.safe.env.sh", shell=True, env={"PATH": "/bin"})
